presto-main/src/test/java/com/facebook/presto/sql/relational/TestLogicalRowExpressions.java

```diff
@@ -121,6 +122,27 @@ public void testDeterminism()
         assertEquals(logicalRowExpressions.filterNonDeterministicConjuncts(expression), or(b, nondeterministic));
     }
 
+    @Test
+    public void testPushNegationToLeaves()
+    {
+        RowExpression a = name("a");
+        RowExpression b = name("b");
+        RowExpression c = name("c");
+        RowExpression d = name("d");
+
+        assertEquals(logicalRowExpressions.pushNegationToLeaves(not(and(a, b))), or(not(a), not(b)));
```

Comment:
Add test cases:
(1) No `not` is in the predicate and see if the output is identical to the input
(2) Consecutive `not` predicate (e.g. `not(not(not(a)))`)
(3) Predicate that does not start with `not`.